### PR TITLE
Support clearing data using Partial Sync

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes https://github.com/realm/realm-java/issues/5758

Partial Sync does not support the `clearTable` instruction. This fixes it. Due to performance concerns, this code path is only triggered for partially synchronized Realms.

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/internal/Table.java [hunk 1]
```diff
@@ -261,10 +261,13 @@ public boolean isEmpty() {
 
     /**
      * Clears the table i.e., deleting all rows in the table.
+     *
+     * WARNING: This method is not supported by Partial Sync, use `findAll().deleteFromRealm`
```
Comment (Zhuinden) [inline realm/realm-library/src/main/java/io/realm/internal/Table.java]: It kind of is supported now, no? With magic 😄

Comment (cmelchior) [reply realm/realm-library/src/main/java/io/realm/internal/Table.java]: The server isn't enforcing it right now. I believe it just ignores the instruction.

Comment (Zhuinden) [reply realm/realm-library/src/main/java/io/realm/internal/Table.java]: No I'm asking because you seem to have just added the means to clear via `isPartial == true` in JNI side

Comment (cmelchior) [reply realm/realm-library/src/main/java/io/realm/internal/Table.java]: There is a subtle difference `table->where().find_all().clear()` creates a query result and delete the individual objects in it. Resulting in a lot of `delete_row` instructions. Which is also why it is slower.

`table->clear()` just results in one `clearTable` instruction, which isn't supported by partial sync right now (mostly because "the whole table" is a loose concept in a world with partial sync.

Comment (nhachicha) [reply realm/realm-library/src/main/java/io/realm/internal/Table.java]: I think  @Zhuinden point was that the comment doesn't makes sense since you implemented internally a fallback to `table->where().find_all().clear()`

I think it should rephrase it to something like 
> WARNING: In the context of partial sync, this method will behave similarly to a `findAll().deleteFromRealm()` which might have a performance impact.

File: realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/PartialSyncTests.java [hunk 2]
```diff
@@ -274,6 +269,30 @@ public void onError(String subscriptionName, Throwable error) {
...
     }
 
+    @Test
+    @RunTestInLooperThread
+    public void clearTable() throws InterruptedException {
+        SyncUser user = UserFactory.createUniqueUser(Constants.AUTH_URL);
+        Realm realm = getPartialRealm(user);
+        looperThread.closeAfterTest(realm);
+
+        // Create test data and make it is uploaded to the server
```
Comment (nhachicha) [inline realm/realm-library/src/syncIntegrationTest/java/io/realm/objectserver/PartialSyncTests.java]: typo? 

Create test data and wait for it to be uploaded to the server
